procrastinate/app.py

```diff
@@ -237,5 +237,8 @@ def migrator(self) -> migration.Migrator:
     def health_check_runner(self) -> healthchecks.HealthCheckRunner:
         return healthchecks.HealthCheckRunner(connector=self.connector)
 
-    async def close_connection_async(self):
-        await self.connector.close_connection()
+    def close_connection(self):
+        self.connector.close_connection()
+
+    def __del__(self):
+        self.close_connection()
```

Comment:
https://github.com/peopledoc/procrastinate/blob/master/procrastinate/cli.py#L109 should be taking care of that, I'd like to understand why it doesn't.